pulp_ansible/app/viewsets.py

```diff
@@ -245,6 +247,20 @@ class CollectionRemoteViewSet(RemoteViewSet):
     queryset = CollectionRemote.objects.all()
     serializer_class = CollectionRemoteSerializer
 
+    def update(self, request, pk, **kwargs):
```

Comment:
I forgot that this `super()` call runs inside a task. Since that's the case we need to do this differently for correctness reasons. I recommend doing these things:

1) Create a new task called `update_collection_remote` and have it contain this update code.
2) Instead of calling super() which would dispatch another task have your `update_collection_remote` task call the `tasks.base.general_update` [in the saw way that the super does](https://github.com/pulp/pulpcore/blob/5a6d5c5f2ebe06922319235b3c845608d10e6f3a/pulpcore/app/viewsets/base.py#L454).
3) You'll need to ensure the `update_collection_remote` task not only locks on the remote being updated but also the AnsibleRepository (this is the main correctness change also)
4) Since you're not calling super have your code return the `OperationPostponed`.
5) Also have you `update_collection_remote` use a Transaction so all postgresql operations happen or none of them do.

Reply:
it may be zero or more than one repository, so I don't think it is possible to lock on repository

Reply:
Why couldn't it just lock on 0 or more also? Pulp supports multi-locking.

Reply:
This looks great